airbyte-cdk/python/unit_tests/sources/declarative/incremental/test_per_partition_cursor.py

```diff
@@ -519,10 +517,19 @@ def test_get_stream_state_includes_parent_state(mocked_cursor_factory, mocked_pa
     assert stream_state == expected_state
 
 
-def test_given_invalid_state_when_set_initial_state_then_raise_config_error(mocked_cursor_factory, mocked_partition_router) -> None:
+def test_per_partition_state_when_set_initial_global_state(mocked_cursor_factory, mocked_partition_router) -> None:
+    first_partition = {"first_partition_key": "first_partition_value"}
+    mocked_partition_router.stream_slices.return_value = [
+        StreamSlice(partition=first_partition, cursor_slice={}),
+    ]
+    mocked_cursor_factory.create.side_effect = [
+        MockedCursorBuilder().with_stream_state({CURSOR_STATE_KEY: "first slice cursor value"}).build(),
+        MockedCursorBuilder().with_stream_state({CURSOR_STATE_KEY: "second slice cursor value"}).build(),
+    ]
     cursor = PerPartitionCursor(mocked_cursor_factory, mocked_partition_router)
 
-    with pytest.raises(AirbyteTracedException) as exception:
-        cursor.set_initial_state({"invalid_state": 1})
-
-    assert exception.value.failure_type == FailureType.config_error
+    cursor.set_initial_state({"global_state_format_key": "global_state_format_value"})
```

Comment:
I'm not sure I understand this test. It feels like if we were to change `cursor.set_initial_state({"global_state_format_key": "global_state_format_value"})` to `cursor.set_initial_state({"states": []})`, the test would still pass which makes me think it isn't testing global state. Should we `assert mocked_cursor_factory.create.calls == [calls(<global state>)] * <number of times we expect it to be called>`?

Reply:
refactored, added test for call_args

Reply:
Has this change been pushed? I don't see it